server/store/memory/peer_store.go

```diff
@@ -88,16 +90,19 @@ func leechersKey(infoHash chihaya.InfoHash) string {
 
 func (s *peerStore) PutSeeder(infoHash chihaya.InfoHash, p chihaya.Peer) error {
 	key := seedersKey(infoHash)
+	i := s.shardIndex(infoHash)
+	if s.shards[i] == nil {
```

Comment:
I think I initially wrote it like this to attempt to save memory, but in practice the allocations probably just slow down the whole program.